proj/src/main/webapp/script.js

```diff
@@ -27,11 +27,64 @@ if(btnLogin){
     });
 }
 
+// Create or join chatroom
+function createOrJoinChat(currentTag){
```

Comment:
There is a lot being done here within anonymous functions. Can you break them out into their own function so it is a bit easier to follow what is going on.

Reply:
Ah, I remember the days when I would put all my logic in a single function nested twelve layers deep.  Alas, those days are gone now....

Reply:
Yeah, I'm not totally sure how this function is supposed to work.  Breaking it up might help.